### PR TITLE
Sparsification update

### PR DESCRIPTION
**Pull request by markurtz**
- update sparsification descriptions and move to preferred verbage
- update classification examples to resnet50

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,80 +50,108 @@ The DeepSparse Engine is a CPU runtime that delivers unprecedented performance b
...
+]
 ```
 
-Great. We can see there are two pruned versions targeting FP32, `conservative` at 100% and `moderate` at >= 99% of baseline accuracy. There is also a `pruned_quant` variant targetting INT8.
+We can see there are two pruned versions targeting FP32 and two pruned, quantized versions targeting INT8.
+The `conservative`, `moderate`, and `aggressive` tags recover to 100%, >=99%, and <99% of baseline accuracy respectively.
 
-Let's say you want to evaluate best performance on FP32 and are okay with a small drop in accuracy, so we can choose `pruned-moderate` over `pruned-conservative`.
+Let's say that we want something that recovers close to the baseline and is very performant, we can choose the pruned_quant-moderate model.
+This model will run [nearly 7 times faster](linktoresnet50example) than the baseline model on a compatible CPU (VNNI instruction set enabled).
```
Comment (bfineran) [inline README.md:126]: placeholder link: `[nearly 7 times faster](linktoresnet50example) `

File: README.md [hunk 2]
```diff
@@ -50,80 +50,108 @@ The DeepSparse Engine is a CPU runtime that delivers unprecedented performance b
...
+    Model(stub=cv/classification/resnet_v1-50/pytorch/sparseml/imagenet-augmented/pruned_quant-aggressive)
+]
 ```
 
-Great. We can see there are two pruned versions targeting FP32, `conservative` at 100% and `moderate` at >= 99% of baseline accuracy. There is also a `pruned_quant` variant targetting INT8.
+We can see there are two pruned versions targeting FP32 and two pruned, quantized versions targeting INT8.
+The `conservative`, `moderate`, and `aggressive` tags recover to 100%, >=99%, and <99% of baseline accuracy respectively.
 
-Let's say you want to evaluate best performance on FP32 and are okay with a small drop in accuracy, so we can choose `pruned-moderate` over `pruned-conservative`.
+Let's say that we want something that recovers close to the baseline and is very performant, we can choose the pruned_quant-moderate model.
```
Comment (bfineran) [inline README.md:125]: word choice: "something"

Comment (bfineran) [inline README.md:125]: should we briefly explain that pruned_quant means pruned and quantized?  Also do we explain what "moderate" means in this context?

File: README.md [hunk 3]
```diff
@@ -50,80 +50,108 @@ The DeepSparse Engine is a CPU runtime that delivers unprecedented performance b
...
-  enabling simple pipelines integration with a few lines of code
-- [Sparsify](https://github.com/neuralmagic/sparsify):
-  Easy-to-use autoML interface to optimize deep neural networks for
-  better inference performance and a smaller footprint
+Sparsification is the process of taking a trained deep learning model and removing redundant information from the over precise and over parameterized network resulting in a faster and smaller model.
+Techniques for sparsification are all encompassing including everything from inducing sparsity using [pruning](https://neuralmagic.com/blog/pruning-overview/) and [quantization](https://arxiv.org/abs/1609.07061) to enabling naturally occurring sparsity using [activation sparsity](http://proceedings.mlr.press/v119/kurtz20a.html) or [winograd/FFT](https://arxiv.org/abs/1509.09308). 
+When implemented correctly, these techniques result in significantly more performant and smaller models with limited to no effect on the baseline metrics.
+For example, pruning plus quantization can give over [7x improvements in performance](resnet50link) while recovering to nearly the same baseline.
```
Comment (bfineran) [inline README.md:58]: "baseline" -> "baseline accuracy"

Comment (bfineran) [inline README.md:58]: nit: placeholder link [7x improvements in performance](resnet50link)

**Review by bfineran — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (markurtz) [reply README.md:126]: waiting on the link to be provided to include here

File: README.md (hunk 2 above)
Comment (markurtz) [reply README.md:125]: In the sentences above this, I list out that it is a pruned, quantized version. And in the intro we link out to what pruning and quantization is

File: README.md (hunk 3 above)
Comment (markurtz) [reply README.md:58]: waiting on the link to be provided to include here

File: README.md [hunk 4]
```diff
@@ -50,80 +50,108 @@ The DeepSparse Engine is a CPU runtime that delivers unprecedented performance b
...
 
-- [SparseZoo](https://github.com/neuralmagic/sparsezoo):
-  Neural network model repository for highly sparse models and optimization recipes
-- [SparseML](https://github.com/neuralmagic/sparseml):
-  Libraries for state-of-the-art deep neural network optimization algorithms,
-  enabling simple pipelines integration with a few lines of code
-- [Sparsify](https://github.com/neuralmagic/sparsify):
-  Easy-to-use autoML interface to optimize deep neural networks for
-  better inference performance and a smaller footprint
+Sparsification is the process of taking a trained deep learning model and removing redundant information from the over precise and over parameterized network resulting in a faster and smaller model.
```
Comment (jeanniefinks) [inline README.md:55]:
```suggestion
Sparsification is the process of taking a trained deep learning model and removing redundant information from the overprecise and over-parameterized network resulting in a faster and smaller model.
```

File: README.md [hunk 5]
```diff
@@ -50,80 +50,108 @@ The DeepSparse Engine is a CPU runtime that delivers unprecedented performance b
...
+Techniques for sparsification are all encompassing including everything from inducing sparsity using [pruning](https://neuralmagic.com/blog/pruning-overview/) and [quantization](https://arxiv.org/abs/1609.07061) to enabling naturally occurring sparsity using [activation sparsity](http://proceedings.mlr.press/v119/kurtz20a.html) or [winograd/FFT](https://arxiv.org/abs/1509.09308). 
+When implemented correctly, these techniques result in significantly more performant and smaller models with limited to no effect on the baseline metrics.
+For example, pruning plus quantization can give over [7x improvements in performance](resnet50link) while recovering to nearly the same baseline accuracy.
+
+The DeepSparse product suite builds on top of sparsification enabling you to easily apply the techniques to your datasets/models using recipe driven approaches.
```
Comment (jeanniefinks) [inline README.md:60]:
```suggestion
The DeepSparse product suite builds on top of sparsification enabling you to easily apply the techniques to your datasets and models using recipe-driven approaches.
```

File: README.md [hunk 6]
```diff
@@ -50,80 +50,108 @@ The DeepSparse Engine is a CPU runtime that delivers unprecedented performance b
...
+Recipes encode the directions for how to sparsify a model into a simple, easily editable format.
+Download a sparsification recipe/sparsified model from the [SparseZoo](https://github.com/neuralmagic/sparsezoo) or create one using [Sparsify](https://github.com/neuralmagic/sparsify), apply it using [SparseML](https://github.com/neuralmagic/sparseml) with only a few lines of code, and deploy with the [DeepSparse Engine](https://github.com/neuralmagic/deepsparse) for unprecedented performance on CPUs.
```
Comment (jeanniefinks) [inline README.md:62]:
```suggestion
- Download a sparsification recipe and sparsified model from the [SparseZoo](https://github.com/neuralmagic/sparsezoo).
- Alternatively, create a recipe for your model using [Sparsify](https://github.com/neuralmagic/sparsify).
- Apply your recipe with only a few lines of code using [SparseML](https://github.com/neuralmagic/sparseml).
- Finally, for GPU-level performance on CPUs, deploy your sparse-quantized model with the [DeepSparse Engine](https://github.com/neuralmagic/deepsparse).


```

File: README.md [hunk 7]
```diff
@@ -50,80 +50,108 @@ The DeepSparse Engine is a CPU runtime that delivers unprecedented performance b
...
+Visualization of the full product flow:
```
Comment (jeanniefinks) [inline README.md:63]:
```suggestion
**Full Deep Sparse product flow:**  
```

File: README.md [hunk 8]
```diff
@@ -50,80 +50,108 @@ The DeepSparse Engine is a CPU runtime that delivers unprecedented performance b
...
+]
 ```
 
-Great. We can see there are two pruned versions targeting FP32, `conservative` at 100% and `moderate` at >= 99% of baseline accuracy. There is also a `pruned_quant` variant targetting INT8.
+We can see there are two pruned versions targeting FP32 and two pruned, quantized versions targeting INT8.
+The `conservative`, `moderate`, and `aggressive` tags recover to 100%, >=99%, and <99% of baseline accuracy respectively.
 
-Let's say you want to evaluate best performance on FP32 and are okay with a small drop in accuracy, so we can choose `pruned-moderate` over `pruned-conservative`.
+Let's say that we want a version of ResNet-50 that recovers close to the baseline and is very performant, we can choose the pruned_quant-moderate model.
+This model will run [nearly 7 times faster](linktoresnet50example) than the baseline model on a compatible CPU (VNNI instruction set enabled).
```
Comment (jeanniefinks) [inline README.md:126]:
```suggestion
This model will run [nearly 7x faster](linktoresnet50example) than the baseline model on a compatible CPU (with the VNNI instruction set enabled).
```

File: README.md [hunk 9]
```diff
@@ -50,80 +50,108 @@ The DeepSparse Engine is a CPU runtime that delivers unprecedented performance b
...
+    Model(stub=cv/classification/resnet_v1-50/pytorch/sparseml/imagenet-augmented/pruned_quant-aggressive)
+]
 ```
 
-Great. We can see there are two pruned versions targeting FP32, `conservative` at 100% and `moderate` at >= 99% of baseline accuracy. There is also a `pruned_quant` variant targetting INT8.
+We can see there are two pruned versions targeting FP32 and two pruned, quantized versions targeting INT8.
+The `conservative`, `moderate`, and `aggressive` tags recover to 100%, >=99%, and <99% of baseline accuracy respectively.
 
-Let's say you want to evaluate best performance on FP32 and are okay with a small drop in accuracy, so we can choose `pruned-moderate` over `pruned-conservative`.
+Let's say that we want a version of ResNet-50 that recovers close to the baseline and is very performant, we can choose the pruned_quant-moderate model.
```
Comment (jeanniefinks) [inline README.md:125]:
```suggestion
For a version of ResNet-50 that recovers close to the baseline and is very performant, choose the pruned_quant-moderate model.
```

File: docs/source/quicktour.md [hunk 10]
```diff
@@ -17,67 +17,90 @@ limitations under the License.
...
+)
+if not engine_sparse.cpu_vnni:
+    print("WARNING: VNNI instructions not detected, quantization speedup not well supported")
+benchmarks_sparse = engine_sparse.benchmark(sample_inputs)
+print(benchmarks_sparse)
+
+print(f"Speedup: {benchmarks_sparse.items_per_second / benchmarks_base.items_per_second:.2f}x")
 ```
 
 ### Quickstart with custom ONNX models
```
Comment (jeanniefinks) [inline docs/source/quicktour.md:106]:
```suggestion
### Quickstart with Custom ONNX Models
```

File: README.md [hunk 11]
```diff
@@ -50,80 +50,108 @@ The DeepSparse Engine is a CPU runtime that delivers unprecedented performance b
...
+)
+if not engine_sparse.cpu_vnni:
+    print("WARNING: VNNI instructions not detected, quantization speedup not well supported")
+benchmarks_sparse = engine_sparse.benchmark(sample_inputs)
+print(benchmarks_sparse)
+
+print(f"Speedup: {benchmarks_sparse.items_per_second / benchmarks_base.items_per_second:.2f}x")
 ```
 
 ### Quickstart with custom ONNX models
```
Comment (jeanniefinks) [inline README.md:162]:
```suggestion
### Quickstart with Custom ONNX Models
```

File: docs/source/index.rst [hunk 12]
```diff
@@ -51,14 +51,26 @@ CPU inference engine that delivers unprecedented performance for sparse models.
...
-fitting seamlessly into your existing deployments as an inference backend.
+The DeepSparse Engine is a CPU runtime that delivers unprecedented performance by taking advantage of natural sparsity within neural networks to reduce compute required as well as accelerate memory bound workloads. It is focused on model deployment and scaling machine learning pipelines, fitting seamlessly into your existing deployments as an inference backend.
 
 `This repository <https://github.com/neuralmagic/deepsparse />`_ includes package APIs along with examples to quickly get started learning about and
 actually running sparse models.
 
+Sparsification
+==============
+
+Sparsification is the process of taking a trained deep learning model and removing redundant information from the over precise and over parameterized network resulting in a faster and smaller model.
```
Comment (jeanniefinks) [inline docs/source/index.rst]:
```suggestion
Sparsification is the process of taking a trained deep learning model and removing redundant information from the overprecise and over-parameterized network resulting in a faster and smaller model.
```

File: docs/source/index.rst [hunk 13]
```diff
@@ -51,14 +51,26 @@ CPU inference engine that delivers unprecedented performance for sparse models.
...
+Techniques for sparsification are all encompassing including everything from inducing sparsity using `pruning <https://neuralmagic.com/blog/pruning-overview/>`_ and `quantization <https://arxiv.org/abs/1609.07061>`_ to enabling naturally occurring sparsity using `activation sparsity <http://proceedings.mlr.press/v119/kurtz20a.html>`_ or `winograd/FFT <https://arxiv.org/abs/1509.09308>`_.
+When implemented correctly, these techniques result in significantly more performant and smaller models with limited to no effect on the baseline metrics.
+For example, pruning plus quantization can give over [7x improvements in performance](resnet50link) while recovering to nearly the same baseline accuracy.
+
+The DeepSparse product suite builds on top of sparsification enabling you to easily apply the techniques to your datasets/models using recipe driven approaches.
```
Comment (jeanniefinks) [inline docs/source/index.rst]:
```suggestion
The Deep Sparse product suite builds on top of sparsification enabling you to easily apply the techniques to your datasets and models using recipe-driven approaches.
```

File: docs/source/index.rst [hunk 14]
```diff
@@ -51,14 +51,26 @@ CPU inference engine that delivers unprecedented performance for sparse models.
...
+Recipes encode the directions for how to sparsify a model into a simple, easily editable format.
+Download a sparsification recipe/sparsified model from the `SparseZoo <https://github.com/neuralmagic/sparsezoo>`_ or create one using `Sparsify <https://github.com/neuralmagic/sparsify>`_, apply it using `SparseML <https://github.com/neuralmagic/sparseml>`_ with only a few lines of code, and deploy with the `DeepSparse Engine <https://github.com/neuralmagic/deepsparse>`_ for unprecedented performance on CPUs.
```
Comment (jeanniefinks) [inline docs/source/index.rst]:
```suggestion
- Download a sparsification recipe and sparsified model from the [SparseZoo](https://github.com/neuralmagic/sparsezoo).
- Alternatively, create a recipe for your model using [Sparsify](https://github.com/neuralmagic/sparsify).
- Apply your recipe with only a few lines of code using [SparseML](https://github.com/neuralmagic/sparseml).
- Finally, for GPU-level performance on CPUs, deploy your sparse-quantized model with the [DeepSparse Engine](https://github.com/neuralmagic/deepsparse).


```

File: docs/source/index.rst [hunk 15]
```diff
@@ -51,14 +51,26 @@ CPU inference engine that delivers unprecedented performance for sparse models.
...
+Visualization of the full product flow:
```
Comment (jeanniefinks) [inline docs/source/index.rst]:
```suggestion
**Full Deep Sparse product flow:**  
```

File: docs/source/quicktour.md [hunk 16]
```diff
@@ -17,67 +17,90 @@ limitations under the License.
...
-Great. We can see there are two pruned versions targeting FP32, 
-`conservative` at 100% and `moderate` at >= 99% of baseline accuracy. 
-There is also a `pruned_quant` variant targeting INT8.
+We can see there are two pruned versions targeting FP32 and two pruned, quantized versions targeting INT8.
+The `conservative`, `moderate`, and `aggressive` tags recover to 100%, >=99%, and <99% of baseline accuracy respectively.
 
-Let's say you want to evaluate best performance on FP32 and are okay with a small drop in accuracy, 
-so we can choose `pruned-moderate` over `pruned-conservative`.
+Let's say that we want a version of ResNet-50 that recovers close to the baseline and is very performant, we can choose the pruned_quant-moderate model.
+This model will run [nearly 7 times faster](linktoresnet50example) than the baseline model on a compatible CPU (VNNI instruction set enabled).
```
Comment (jeanniefinks) [inline docs/source/quicktour.md:75]:
```suggestion
This model will run [nearly 7x faster](linktoresnet50example) than the baseline model on a compatible CPU (with the VNNI instruction set enabled).
```

File: README.md [hunk 17]
```diff
@@ -50,80 +50,108 @@ The DeepSparse Engine is a CPU runtime that delivers unprecedented performance b
...
 outputs, inference_time = engine.timed_run(inputs)
 ```
 
-**MobileNetV1 Optimized**
+**ResNet-50 Sparsified**
 
 When exploring available optimized models, you can use the `Zoo.search_optimized_models` utility to find models that share a base.
 
-Let us try this on the dense MobileNetV1 to see what is available.
+Let us try this on the dense ResNet-50 to see what is available.
```
Comment (jeanniefinks) [inline README.md:105]:
```suggestion
Try this on the dense ResNet-50 to see what is available:
```

File: docs/source/quicktour.md [hunk 18]
```diff
@@ -17,67 +17,90 @@ limitations under the License.
...
 
-**MobileNetV1 Optimized**
+**ResNet-50 Sparsified**
 
-When exploring available optimized models, you can use the `Zoo.search_optimized_models` 
-utility to find models that share a base. 
+When exploring available optimized models, you can use the `Zoo.search_optimized_models` utility to find models that share a base.
 
-Let us try this on the dense MobileNetV1 to see what is available.
+Let us try this on the dense ResNet-50 to see what is available.
```
Comment (jeanniefinks) [inline docs/source/quicktour.md:49]:
```suggestion
Try this on the dense ResNet-50 to see what is available:
```

File: docs/source/quicktour.md [hunk 19]
```diff
@@ -17,67 +17,90 @@ limitations under the License.
...
 
-Great. We can see there are two pruned versions targeting FP32, 
-`conservative` at 100% and `moderate` at >= 99% of baseline accuracy. 
-There is also a `pruned_quant` variant targeting INT8.
+We can see there are two pruned versions targeting FP32 and two pruned, quantized versions targeting INT8.
+The `conservative`, `moderate`, and `aggressive` tags recover to 100%, >=99%, and <99% of baseline accuracy respectively.
 
-Let's say you want to evaluate best performance on FP32 and are okay with a small drop in accuracy, 
-so we can choose `pruned-moderate` over `pruned-conservative`.
+Let's say that we want a version of ResNet-50 that recovers close to the baseline and is very performant, we can choose the pruned_quant-moderate model.
```
Comment (jeanniefinks) [inline docs/source/quicktour.md:74]:
```suggestion
For a version of ResNet-50 that recovers close to the baseline and is very performant, choose the pruned_quant-moderate model.
```

**Review by bfineran — APPROVED** [review]

**Review by jeanniefinks — APPROVED** [review]

**Review by NMsasa — APPROVED** [review]

**Pull request merged**